framework/src/utils/MortarUtils.C

```diff
@@ -222,11 +222,20 @@ projectQPoints3d(const Elem * const msm_elem,
         x1 += Moose::fe_lagrange_2D_shape(sub_element_type(), FIRST, n, xi) *
               primal_elem->point(sub_elem_node_indices[n]);
       auto u = x1 - x0;
+
       VectorValue<Dual2> F(u(1) * normal(2) - u(2) * normal(1),
                            u(2) * normal(0) - u(0) * normal(2),
                            u(0) * normal(1) - u(1) * normal(0));
 
-      if (MetaPhysicL::raw_value(F).norm() < 1e-12)
+      Real projection_tolerance(1e-10);
+
+      // Normalize tolerance with quantities involved in the projection.
+      u(0) += 1.0e-36;
```

Comment:
This number seems so arbitrary. How did you pick it?

Reply:
Also this comment seems to apply to the later lines. Maybe group these lines together with no spaces

Reply:
Added that number to prevent having a divide-by-zero in the vector norm. The delta should be a good number of orders of magnitude less than the tolerance we use to solve the projection problem while still avoiding the numerical issue. But, other than that, it's arbitrary, based on what worked in the past. We have at least another instance in the code of the same usage: What do you suggest?

Reply:
I mostly meant the `36` part of it. I do wish we didn't have so many different tolerances in mortar, but as long as things are "working" or tolerances are adjusted so that "more things work", then that is a good enough reason and guide for how the tolerances should be set...? I just struggle with seeming arbitrariness is all. You're no more guilty of it than @jwpeterson or I am in the history of our mortar capability

Reply:
So what you are saying is that you are intolerant of... tolerances? #dadjokes

Reply:
Haha nailed it!